apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginCfg.java

```diff
@@ -12,23 +10,30 @@
 public enum PluginCfg {
     INSTANCE;
 
-    private List<String> pluginClassList = new ArrayList<String>();
+    private List<PluginDefine> pluginClassList = new ArrayList<PluginDefine>();
+    private List<String> disabledPlugins = new ArrayList<String>();
 
     void load(InputStream input) throws IOException {
         try {
             BufferedReader reader = new BufferedReader(new InputStreamReader(input));
-            String pluginDefineClassName = null;
-            while ((pluginDefineClassName = reader.readLine()) != null) {
-                if (!StringUtil.isEmpty(pluginDefineClassName)) {
-                    pluginClassList.add(pluginDefineClassName.trim());
+            String pluginDefine = null;
+            while ((pluginDefine = reader.readLine()) != null) {
+                PluginDefine plugin = PluginDefine.build(pluginDefine);
+                if (!plugin.disabled(disabledPlugins)) {
```

Comment:
Are you sure this is going to work? 

According to `premain`
```java
    public static void premain(String agentArgs, Instrumentation instrumentation) throws PluginException {
        SnifferConfigInitializer.initialize();

        final PluginFinder pluginFinder = new PluginFinder(new PluginBootstrap().loadPlugins());

        ServiceManager.INSTANCE.boot();
```

`load(InputStream input)` invokes before `ServiceManager.boot`. **disabledPlugins**(initialize in `DisabledPluginInitService`) must be empty every time, how does this pr work?